scripts/mkcloud

```diff
@@ -540,12 +540,36 @@ function setupadmin()
     fi
 }
 
+function wait_for_node_shutdown()
+{
+    wait_for 150 1 "virsh domstate $1 | grep shut.off" '$1 node to shut down'
+}
+
+function remove_snapshot_volume()
+{
+    if [[ "${1}" =~ \.snap$ ]]; then
```

Comment:
> Only if the path $1 does not contain whitespace.

Nope.
Double brackets `[[ ]]` are intelligent, they know that this is just one variable, so it also works with whitespace and even with empty string. The goal of using `[[` is to reduce code and make it more readable. So we should use the features it offers.

See:

```
foo="foo bar"
[[ $foo =~ foo ]] && echo works
[[ $foo =~ bar ]] && echo works
[[ $foo =~ "foo bar" ]] && echo works # only here you need to quote the string, but not the variable
```